endpoint/shellComp/shell.c

```diff
@@ -0,0 +1,496 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "shell.h"
+
+// Legato headers
+#include "interfaces.h"
+#include "legato.h"
+
+// Linux headers
+#include <errno.h>    // Error integer and strerror() function
+#include <fcntl.h>    // Contains file controls like O_RDWR
+#include <termios.h>  // Contains POSIX terminal control definitions
+#include <unistd.h>   // write(), read(), close()
+
+// cURL headers
+#include <curl/curl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "common/ta_errors.h"
+
+int sh_ok(char **argv);
+int sh_help(char **argv);
+int sh_ecm(char **argv);
+int sh_radio(char **argv);
+int sh_ping(char **argv);
+int sh_dns(char **argv);
+int sh_route(char **argv);
+int sh_at(char **argv);
+int sh_transaction(char **argv);
+int sh_connect(char **argv);
+int sh_exit(char **argv);
+
+static int sh_launch(char **args);
+int open_serialport(char *dev, speed_t baudrate);
+
+/* List of builtin commands */
+static command builtin_cmd[] = {
+    {"ok", "OK", sh_ok},
+    {"help", "print the help message", sh_help},
+    {"ecm", "enable/disable ssh login [on/off]", sh_ecm},
```

Comment:
Why `ecm` here?

Reply:
Ecm is the ssh login interface.

Reply:
You have to append the top-level documentation accordingly.